packages/flutter/lib/src/widgets/text.dart

```diff
@@ -210,16 +215,40 @@ class Text extends StatelessWidget {
     this.textScaleFactor,
     this.maxLines,
   }) : assert(data != null),
+       span = null,
        super(key: key);
 
+  /// Creates a text widget from a [TextSpan].
+  const Text.fromSpan(this.span, {
+    Key key,
+    this.textAlign,
+    this.textDirection,
+    this.softWrap,
+    this.overflow,
+    this.textScaleFactor,
+    this.maxLines,
+  }): assert(span != null),
+      data = null,
+      style = null,
+      super(key: key);
+
   /// The text to display.
+  ///
+  /// If constructed with the default constructor, `span` will be null.
   final String data;
 
+  /// The text to display as a span.
+  ///
+  /// If constructed from a [TextSpan], `data` will be null.
+  final TextSpan span;
+
   /// If non-null, the style to use for this text.
   ///
   /// If the style's "inherit" property is true, the style will be merged with
   /// the closest enclosing [DefaultTextStyle]. Otherwise, the style will
   /// replace the closest enclosing [DefaultTextStyle].
+  ///
+  /// If constructed from a [TextSpan], the style will be null.
```

Comment:
we should probably allow style to be applied when you provide a TextSpan, the same way we do with text, just for completeness.

Reply:
Done